frontend/src/metabase/query_builder/containers/QueryBuilder.jsx

```diff
@@ -398,22 +418,45 @@ function QueryBuilder(props) {
     setIsShowingToaster(false);
   }, []);
 
+  const isLocationAllowed = useCallback(
+    location => {
+      if (!question?.isNative()) {
+        return true;
+      }
```

Comment:
This condition will change as the epic moves forward and should be gone after milestone 3.